resources/git-fork.md

```diff
@@ -0,0 +1,98 @@
+![Forking on GitHub](images/github-fork.png)
+
+## Why would we want a fork?
+### Why not just branch?
+
+So far we have mostly discussed the reasons one may want or need to create a _hard_ fork of some project, where we intend on making permanently incompatible changes, but we have not justified our more typical (soft) fork situation. As you recall, this is for work which we want to eventually merge back into the main repository. Some reasons for doing this are:
+
+* **User workspaces:** when projects involve a large number of developers (imagine a popular open source or large enterprise project), if each developer were pushing branches for their work the repository would become cluttered with many branches mostly of interest to only one person. When you are synchronizing your local instance, you have many more branches listed than you need because you may see branches for everybody's work.
+* **Clean main repository:** as in the above case, but even for smaller teams, often the main repository is kept as the source-of-truth for the `master` (or `develop`) branch, and is otherwise reserve for CI and CD tasks, such as preserving production release versions. Branches for work-in-progress are relegated to user forks only in order to keep the central repo tidy.
+* **Experimental changes:** Sometimes you need to make large architectural changes or adjust CI/CD configuration, and it isn't acceptable to block work on the main repo. Configuring another pipeline against a fork is one option to get around this.
+
+## Dealing with multiple `remote`s
+When you set up your local repository with multiple `remote` repositories (usually the main repository and your own fork), there are slight changes to some of the `git` commands you are accustomed to using. Specifically, for any commands that deal with remote repositories (`push`, `pull`, `fetch` etc), you now need to specify the repository with which you want to interact.
+
+For the following examples, imagine you cloned from your fork, so the repo copy under your account is the remote named `origin`, and we've added a second remote called `upstream` for the main reposiory.
+
+### `pull`
+When pulling the latest changes for a tracked remote branch, you now specifiy the full source, including the remote name:
+```shell
+$> git pull upstream master
+```
+
+### `push`
+Now when you push a branch, you will nee to specifiy the target remote by name:
+```shell
+$> git push origin my-cool-feature
+```
+A common workflow is to push your work-in-progress feature branches to your fork, and then issue a Pull Request from there back to the main repository.
+
+### `fetch`
+If you want to retrieve all of the latest changes from some remote (usually `upstream`), you can do so via `fetch`:
+```shell
+$> git fetch upstream
+```
+
+### `rebase`
+_TBD_ - should we cover this?
+
+## The Short Version
```

Comment:
Should we add a link to this section near the top of this file?

Reply:
Good idea, added.